factory_client.py

```diff
@@ -149,7 +149,8 @@ def get_target_system_image(self, target: Target, out_dir: str, progress: Progre
             logger.info('Downloading Target system image...; Target: {}, image: {}'
                         .format(target.name, image_filename))
 
-            image_resp = http_get(image_url, headers=self._auth_headers)
+            image_resp = requests.get(image_url, headers=self._auth_headers)
+            image_resp.raise_for_status()
```

Comment:
Can you share how the program output will differ. The way it was looks like you get roughly the same thing:
 https://github.com/foundriesio/ci-scripts/blob/ff4e797d5fd005ddb37bcea3a6a8efcc0364f168/helpers.py#L151

Reply:
I need a status code of a response to catch `404`. `raise_for_status` adds a response object to the HTTPError context while `http_get` doesn't do it.
An alternative to this, could be altering the `http_get` implementation so it adds a response object or just a status code to an exception context.
I am fine with either options.

Reply:
ah - i see now. the caller of this function is catching the error.